File: leetcode/0509. 斐波那契数/main.go
package main

import "fmt"

func main()  {
	fmt.Println(fib(5))
}

func fib(n int) int {
	if n < 2 {
		return n
	}
	total, pre, post := 0, 0, 1
	for i := 1; i < n; i ++ {
		total = pre + post
		pre = post
		post = total
	}
	return total
}

File: leetcode/0509. 斐波那契数/main.ts
function fib(n: number): number {
    if (n < 2) {
        return n
    }
    let total: number = 0
    let pre: number = 0
    let curr: number = 1
    for (let i = 1; i < n; i++) {
        total = pre + curr
        pre = curr
        curr = total
    }
    return total
};

console.log(fib(3))